### PR TITLE
Create FAQ section and Accordion Component

### PR DESCRIPTION
**Pull request by teoh4770**
Create FAQ Section that contain a FAQ heading and an accordion

Checklist:

- [X]  Semantic Accordion Component
- [X]  Test with placeholder text
- [X]  Styling

### CODE REVIEW

File: src/components/sections/FAQ.section.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      answers={faq.answers}
+    />
+  ));
+
+  return (
+    <section className="bg-slate-950 text-white">
+      <h2 className="mb-10 text-center text-6xl font-bold uppercase">
+        Frequently Asked <span className="text-mint">Questions</span>
+      </h2>
+      <Accordion>{AccordionItems}</Accordion>
```
Comment (juancwu) [inline src/components/sections/FAQ.section.tsx:33]: You would add a wrapper here to place the Accordion how it supposed to look like in the current page instead of having the component itself know the layout of where it is going to be.

File: src/components/sections/FAQ.section.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { Accordion, AccordionItem } from "@components";
+
+const FAQSection = () => {
+  const faqs = [
```
Comment (juancwu) [inline src/components/sections/FAQ.section.tsx:4]: I would put `faqs` outside of the component since they are static information.

File: src/components/sections/FAQ.section.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    {
+      question: "🕒 Cool! When and where is HawkHacks happening?",
+      answers: ["HawkHacks will take place for 36 hours, starting on May 13th, Friday @ 6PM EDT all the way till  May 15th, Sunday @ 2PM EDT.", "HawkHacks is a fully virtual event, meaning everything will be hosted and managed through Hopin, Discord, and Twitch!"],
+    },
+    {
+      question: "💰 That sounds good and all, but how much does it cost?",
+      answers: ["Nothing! Free! Nada! Due to our amazing sponsors and support systems, attending and participating at HawkHacks is absolutely free!"],
+    },
+  ];
```
Comment (juancwu) [inline src/components/sections/FAQ.section.tsx:17]:
```suggestion
  const faqs = [
    {
      question: "🤔 So, what exactly is a hackathon?",
      answers: ["A hackathon is made up of two sections - a social coding event where programmers, designers, and developers collaborate in teams to solve a problem. You can either create that falls under a category hack, or make whatever your heart desires as a general hack.", "The second section is networking events, seminars, and presentations that happen throughout the hackathon conducted by industry and academic professionals - if you’ve been putting off learning that one programming language for a while, now might be the time to learn!"],
    },
    {
      question: "🕒 Cool! When and where is HawkHacks happening?",
      answers: ["HawkHacks will take place for 36 hours, starting on May 13th, Friday @ 6PM EDT all the way till  May 15th, Sunday @ 2PM EDT.", "HawkHacks is a fully virtual event, meaning everything will be hosted and managed through Hopin, Discord, and Twitch!"],
    },
    {
      question: "💰 That sounds good and all, but how much does it cost?",
      answers: ["Nothing! Free! Nada! Due to our amazing sponsors and support systems, attending and participating at HawkHacks is absolutely free!"],
    },
  ];

const FAQSection = () => {
```

**Review by juancwu — CHANGES_REQUESTED** [review]

File: src/components/Accordion/AccordionItem.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                }`}
+              ></span>
+            </div>
+          </div>
+        </button>
+      </h2>
+      <div
+        id={`panel-content-${id}`}
+        className="grid font-light"
+        style={showAnswerCSS}
```
[the remark this answers is not in the document]
Comment (teoh4770) [reply src/components/Accordion/AccordionItem.tsx:75]: let me try again

[the remark this answers is not in the document]
Comment (teoh4770) [reply src/components/Accordion/AccordionItem.tsx:75]: alright, the style class is removed

[the remark this answers is not in the document]
Comment (teoh4770) [reply src/components/Accordion/AccordionItem.tsx:75]:
```
className={`grid font-light transition-[grid-template-rows] duration-500 ${
          open ? "grid-rows-[0fr]" : "grid-rows-[1fr]"
        }`}
```

File: src/components/Accordion/Accordion.tsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { FC, PropsWithChildren } from "react";
+
+const Accordion: FC<PropsWithChildren> = ({ children }) => {
+  return <div className="mx-auto grid max-w-[68.75rem]">{children}</div>;
```
[the remark this answers is not in the document]
Comment (teoh4770) [reply src/components/Accordion/Accordion.tsx:4]: yeah, understood

[the remark this answers is not in the document]
Comment (teoh4770) [reply src/components/Accordion/Accordion.tsx:4]:
```
const Accordion: FC<PropsWithChildren> = ({ children }) => {
  return <div className="w-full">{children}</div>;
};
```

File: src/components/sections/FAQ.section.tsx (hunk 2 above)
Comment (teoh4770) [reply src/components/sections/FAQ.section.tsx:4]: basically you want to prevent the variable from re-render?

Comment (teoh4770) [reply src/components/sections/FAQ.section.tsx:4]:
```
const faqs = [
  {
    question: "🤔 So, what exactly is a hackathon?",
    answers: [
      "A hackathon is made up of two sections - a social coding event where programmers, designers, and developers collaborate in teams to solve a problem. You can either create that falls under a category hack, or make whatever your heart desires as a general hack.",
      "The second section is networking events, seminars, and presentations that happen throughout the hackathon conducted by industry and academic professionals - if you’ve been putting off learning that one programming language for a while, now might be the time to learn!",
    ],
  },
  {
    question: "🕒 Cool! When and where is HawkHacks happening?",
    answers: [
      "HawkHacks will take place for 36 hours, starting on May 13th, Friday @ 6PM EDT all the way till  May 15th, Sunday @ 2PM EDT.",
      "HawkHacks is a fully virtual event, meaning everything will be hosted and managed through Hopin, Discord, and Twitch!",
    ],
  },
  {
    question: "💰 That sounds good and all, but how much does it cost?",
    answers: [
      "Nothing! Free! Nada! Due to our amazing sponsors and support systems, attending and participating at HawkHacks is absolutely free!",
    ],
  },
];

const FAQSection = () => {
```

File: src/components/Accordion/AccordionItem.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
+import { useState } from "react";
+
+interface AccordionValue {
```
Comment (juancwu) [inline src/components/Accordion/AccordionItem.tsx:3]:
```suggestion
interface AccordionItemProps {
```

File: src/components/Accordion/AccordionItem.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  const buttonHandler = () => {
+    return setOpen((open) => !open);
+  };
+
+  return (
+    <div
+      className={`accordion-item  ${
+        id === props.answers.length ? "border-y" : "border-t"
+      }`}
+      key={id}
```
Comment (juancwu) [inline src/components/Accordion/AccordionItem.tsx:38]:
```suggestion
```

File: src/components/Accordion/AccordionItem.tsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+interface AccordionValue {
+  id: number;
+  question: string;
+  answers: string[];
+}
+
+const AccordionItem = (props: AccordionValue) => {
+  const [open, setOpen] = useState(false);
+
+  const id = props.id;
```
Comment (juancwu) [inline src/components/Accordion/AccordionItem.tsx:12]:
```suggestion
```

File: src/components/Accordion/AccordionItem.tsx [hunk 9]
```diff
@@ -0,0 +1,79 @@
+import { useState } from "react";
+
+interface AccordionValue {
+  id: number;
```
Comment (juancwu) [inline src/components/Accordion/AccordionItem.tsx:4]:
```suggestion
```

File: src/components/Accordion/Accordion.tsx [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+      "Nothing! Free! Nada! Due to our amazing sponsors and support systems, attending and participating at HawkHacks is absolutely free!",
+    ],
+  },
+];
+
+const Accordion = () => {
+  const AccordionItems = faqs.map((faq, index) => (
+    <AccordionItem
+      key={index}
+      id={index}
```
Comment (juancwu) [inline src/components/Accordion/Accordion.tsx:30]:
```suggestion
```

I don't think it needs an id

**Review by juancwu — CHANGES_REQUESTED** [review]

File: src/components/Accordion/Accordion.tsx (hunk 10 above)
Comment (teoh4770) [reply src/components/Accordion/Accordion.tsx:30]: yeah, without an id would make it more clean for sure

**Review by juancwu — APPROVED** [review]
